src/main/Config.h

```diff
@@ -212,6 +212,9 @@ class Config : public std::enable_shared_from_this<Config>
     // fees, and scp history in the database
     bool MODE_STORES_HISTORY;
 
+    // A config parameter that stores ledger headers in the database
```

Comment:
Extend this comment to include the fact that this exists in order to support a sub-mode of in-memory mode, where `MODE_STORES_HISTORY` is false but `MODE_STORES_HISTORY_LEDGERHEADERS` remains true. That is: that there is no legal configuration where `MODE_STORES_HISTORY_LEDGERHEADERS` is false but `MODE_STORES_HISTORY` is true. Or in yet other words, that only 3 of the 4 states of these 2 flags are legal.

(Also option to discuss here: I know we've gone back and forth on orthogonal flags vs. enumerated modes, but maybe this is as good as any to ask whether to reconsider merging `MODE_USES_IN_MEMORY_LEDGER`, `MODE_STORES_HISTORY` and `MODE_STORES_HISTORY_LEDGERHEADERS` back into a single enum. I'm not sure which of the _8_ possible combinations is legal off the top of my head, which is not a great sign.)

Reply:
mmm, I don't know @graydon : we used to have those "uber flags" used in the code base and it was actually very hard to understand which parts of the code had to change when touching things.

What we may want to do in this commit is probably change `MODE_STORES_HISTORY` to something that doesn't overlap with `MODE_STORES_HISTORY_LEDGERHEADERS` (right now it's hard to tell how they relate to each other: are they mutually exclusive?!).

So if we rename `MODE_STORES_HISTORY` to something like `MODE_STORES_HISTORY_MISC` (and introduce `MODE_STORES_HISTORY_LEDGERHEADERS` before) we can make it clear that they complement each other.

We can add helper functions to help check if *any* historical data is enabled (that seems to be the common case), or maybe to enable/disable "persist all historical data" (to flip both at the same time).

Reply:
I like the idea to rename `MODE_STORES_HISTORY`. I ended up implementing `modeStoresAllHistory` and `modeStoresAnyHistory`, and that seems to improve clarity quite a bit.

Reply:
looking good!